docs/qubit-routing/qaoa-qubit-routing.md

```diff
@@ -0,0 +1,125 @@
+
+![qaoa_routing_example_2](/img/routing_2.png) ![routing_example_circuit_2](/img/swap_ckts.png)
+
+## Special properties of QAOA circuits
+As mentioned earlier, QAOA circuits consists of alternating layers of a cost and a mixer layer. 
+
+For a standard QAOA, the mixer layer consists of single-qubit $RX$ gates. Therefore, routing is only required for the cost layer, where each gate in the cost layer is either a parameterized $RZZ(\gamma)$ or $RZ(\gamma)$. Since all gates within the cost layer commute with each other, our qubit routing solution can take advantage of this by reordering the gates for better SWAP sequences and lower circuit depth. 
+
+Before, we start desigining a solution to implement qubit routing, note that we can also omit the single qubit $RZ(\gamma)$ gates as they do not influence the routing solution. Moreover, as the remaining gates in the sub-circuit to SWAP are all $RZZ(\gamma)$, we can reformulate the set of gates as a graph. One can simlpify even further to represent the same as a list of edges of the graph. This transformation is also depicted in the figures below. 
+
+![problem_representation](/img/problem_representation.png)
+
+
+## Qubit Routing solutions in OpenQAOA
+Implemeting your own custom qubit routing solution in OpenQAOA is really simple! Owing to the simplifications in the input circuit from QAOA (as discussed above), it can be represented as a list of edges. 
+The custom routing solution is then expected to accept the circuit (as a list of edges) and the `openqaoa.DeviceBase` object as input to produce a routed circuit.
+
+The custom routing function is also expected to output other relevant information regarding the process. These are described below:
+
+- initial_mapping (`List[int]`): This list reports selected qubits on the QPU that the algorithm will be executed on. For example, it is assumed that for an `initial_mapping = [b,a,d,c,f,e]`, the logical to physical qubit mapping is defined as `{0:b,1:a,2:d,3:c,4:f,5:e}`
+- routed_gates (`List[List[int]]`): The routed gate set is output in the same format as the input gate set. This list contains pairs of qubits where a two-qubit gates is applied. Each of these pair is either an `RZZ` or a `SWAP` gate. Each pair is defined using logical qubits, i.e. qubits numbered from `[0, 1, 2, ... n-1]`
+- swap_mask (`List[bool]`): This output tags each qubit pair (gate) in the list above as either a `SWAP` gate if the entry is `True` or an `RZZ` gate if the entry is `False`.
+- final_qubit_mapping (`List[int]`): This list reports the final order of the shuffled qubits after the application of `SWAPs`. The list must be labelled with qubit indices from `0` to `n-1`
```

Comment:
This list reports the final order of shuffled qubits -> the qubit indices being referred to here is the logical qubit numbering right? As in this is a mapping of how the logical qubit numbering changes at the end of the whole circuit?

Reply:
yes, correct